### ISSUE TITLE
Local actions in .github/actions/ are not checked

### ISSUE THREAD

**Issue by benoitchantre** [dependabot/dependabot-core#6345]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

github-actions

### dependabot.yml content

version: 2

updates:
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "weekly"

### What you expected to see, versus what you actually saw

When directory is set to `"/"`, I expect that local actions located in `.github/actions/` are also checked.

This is not the case.

You need to add a new item in the config file to allow Dependabot to check a specific action

```
  - package-ecosystem: "github-actions"
    directory: "/.github/actions/deployment-tools"
    schedule:
      interval: "weekly"
```

**Comment by deivid-rodriguez** [comment]
Hei @benoitchantre! Did you find `.github/actions` documented anywhere? I did find one mention at https://docs.github.com/en/actions/creating-actions/about-custom-actions#choosing-a-location-for-your-action, is that the advice you followed when you put your actions at a `.github/actions` folder?

**Comment by benoitchantre** [comment]
Yes, the location I choosed was influenced by the documentation.

> If you plan to combine action, workflow, and application code in a single repository, we recommend storing actions in the .github directory. For example, .github/actions/action-a and .github/actions/action-b.

Source: https://docs.github.com/en/actions/creating-actions/about-custom-actions#choosing-a-location-for-your-action

**Comment by Jason3S** [comment]
Moving comment from:
https://github.com/dependabot/dependabot-core/issues/7495#issuecomment-1729129950

> I have been running into the same issue.
> 
> I took a look at the [code](https://github.com/dependabot/dependabot-core/blob/main/github_actions/lib/dependabot/github_actions/file_fetcher.rb). It also looks for `action.yml` and `action.yaml` in the directory specified. Which means, there is workaround, a bit painful, but it works.
> 
> `.github/dependabot.yml`
> 
> ```yaml
>   - package-ecosystem: "github-actions"
>     # Workflow files stored in the
>     # default location of `.github/workflows`
>     directory: "/.github/actions/a_reusable_workflow"
>     schedule:
>       interval: "daily"
> ```
> 
> It means you will need to add an entry in `dependabot.yml` for every single `.github/actions` folder.

@deivid-rodriguez 

The change to add support for scanning `.github/actions` does not look too bad. Should we look for just `action.*(yaml|yml)` files or all `*.(yaml|yml)` files in `.github/actions`?

https://github.com/dependabot/dependabot-core/blob/0997935bfe7531a9f623c432eb3004846b369621/github_actions/lib/dependabot/github_actions/file_fetcher.rb#L48-L67

**Comment by deivid-rodriguez** [comment]
I guess just `action.ya?ml` for now? It seems that's the documented name at https://docs.github.com/en/actions/creating-actions/about-custom-actions#types-of-actions.

**Comment by ia0** [comment]
Any update on this? The workaround is particularly annoying because it seems using a glob doesn't work:

```
  - package-ecosystem: "github-actions"
    directories:
      - "/.github/workflows"
      - "/.github/actions/*"
```

**Comment by ssbarnea** [comment]
The problem is that the actions schema is different enough from the workflows ones. They share a lot of similarities but they are not compatible. The code needs to be able to read and update both.

**Comment by ia0** [comment]
> The problem is that the actions schema is different enough from the workflows ones. They share a lot of similarities but they are not compatible. The code needs to be able to read and update both.

I don't think this is the problem of this issue. This issue is about needing to be explicit about local action paths. If the problem was a schema issue, then it shouldn't work to check local actions at all, regardless of how they are specified (which was #4327).

**Comment by Jason3S** [comment]
There is a work around as described above.
I wrote an action that will update the `.github/dependabot.yml` file to include files found in `.github/actions`: https://github.com/streetsidesoftware/actions/tree/main/public/update-dependabot-github-actions

Here is an example of where it is used: https://github.com/streetsidesoftware/cspell-dicts/blob/main/.github/workflows/update-dependabot.yml

**Comment by gionn** [comment]
Now it should be supported using the new `directories` property together with wildcard:

```yml
version: 2
updates:
  - package-ecosystem: "github-actions"
    directories:
      - "/"
      - "/.github/actions/*"
    schedule:
      interval: "monthly"
```

except if you have another yaml into one of the actions folder:
https://github.com/dependabot/dependabot-core/issues/13241

verified in:
https://github.com/Alfresco/alfresco-ansible-deployment/pull/1284

**Comment by Jason3S** [comment]
> Now it should be supported using the new `directories` property together with wildcard:
>
> ```
> version: 2
> updates:
>   - package-ecosystem: "github-actions"
>     directories:
>       - "/"
>       - "/.github/actions/*"
>     schedule:
>       interval: "monthly"
> ```
>
> except if you have another yaml into one of the actions folder: [#13241](https://github.com/dependabot/dependabot-core/issues/13241)
> 
> verified in: [Alfresco/alfresco-ansible-deployment#1284](https://github.com/Alfresco/alfresco-ansible-deployment/pull/1284)

That is awesome! Thank you.``

**Comment by ahus1** [comment]
I tried the solution - or rather more than a workaround - to specify the actions via "directories" as "/" and "/.github/actions/*". This now leads to multiple PRs: 
* One PR to update all the workflows
* One PR for each shared action 

While this is kind of better than before (eventually all actions are updated), this is now problematic as I will not have one CI build that tests the whole update. Maybe it would even break the upgrade as it needs to be updated in all the places to have the correct result. 

So I'd say this is is just an incomplete workaround. 

<img width="1009" height="530" alt="Image" src="https://github.com/user-attachments/assets/81efc432-c18c-4f36-849a-0f9fcf0e7a8a" />

**Comment by awoimbee** [comment]
> This now leads to multiple PRs

Just create a group

```yaml
  - package-ecosystem: "github-actions"
    directories:
      - "/"
      - "/.github/actions/*"
    schedule:
      interval: "monthly"
    groups:
      actions-deps:
        patterns:
          - "*"
    commit-message:
      prefix: deps
    cooldown:
      default-days: 5
```
